### ISSUE TITLE
[Issue]: Error when using IPEX: %1 is not a valid Win32 application.

### ISSUE THREAD

**Issue by jacob1218** [vladmandic/automatic#2255]
### Issue Description

Tried running IPEX after OpenVINO took over 15s per iteration, didn't work.

### Version Platform Description

Windows 11 22H2
SD.Next 89ba8e3c
Intel ARC A770 16GB

### Relevant log output

```shell
:10:31-523095 INFO     Starting SD.Next
17:10:31-529619 INFO     Python 3.10.11 on Windows
17:10:31-701360 INFO     Version: app=sd.next updated=2023-09-20 hash=89ba8e3c
                         url=https://github.com/vladmandic/automatic.git/tree/master
17:10:31-754789 INFO     Platform: arch=AMD64 cpu=AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD system=Windows
                         release=Windows-10-10.0.22621-SP0 python=3.10.11
17:10:31-757796 INFO     Intel OneAPI Toolkit detected
17:10:31-773309 ERROR    Intel OneAPI Toolkit is not activated! Activate OneAPI manually!
17:10:31-774814 INFO     Installing package: torch==2.0.0a0 intel_extension_for_pytorch==2.0.110+gitba7f6c1 -f
                         https://developer.intel.com/ipex-whl-stable-xpu
17:11:28-488704 INFO     Extensions: disabled=[]
17:11:28-490703 INFO     Extensions: enabled=['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser',
                         'stable-diffusion-webui-rembg', 'SwinIR'] extensions-builtin
17:11:28-501786 INFO     Extensions: enabled=[] extensions
17:11:28-503785 INFO     No changes detected: Quick launch active
17:11:28-505294 INFO     Verifying requirements
17:11:28-507299 INFO     Installing package: addict
17:11:29-451751 INFO     Installing package: aenum
17:11:30-576703 INFO     Installing package: aiohttp
17:11:33-434471 INFO     Installing package: anyio
17:11:34-935732 INFO     Installing package: appdirs
17:11:35-887322 INFO     Installing package: astunparse
17:11:37-196091 INFO     Installing package: blendmodes
17:11:39-421848 INFO     Installing package: clean-fid
17:12:21-324891 INFO     Installing package: easydev
17:12:22-972469 INFO     Installing package: extcolors
17:12:24-164010 INFO     Installing package: facexlib
17:12:47-781681 INFO     Installing package: filetype
17:12:49-024931 INFO     Installing package: future
17:12:51-643712 INFO     Installing package: gdown
17:12:53-441440 INFO     Installing package: gfpgan
17:13:11-358841 INFO     Installing package: GitPython
17:13:13-448627 INFO     Installing package: httpcore
17:13:15-218544 INFO     Installing package: inflection
17:13:16-539045 INFO     Installing package: jsonmerge
17:13:18-632271 INFO     Installing package: kornia
17:13:22-802748 INFO     Installing package: lark
17:13:24-452906 INFO     Installing package: lmdb
17:13:25-772472 INFO     Installing package: lpips
17:13:27-173787 INFO     Installing package: omegaconf
17:13:29-213183 INFO     Installing package: open-clip-torch
17:13:36-623571 INFO     Installing package: opencv-contrib-python-headless
17:13:39-185448 INFO     Installing package: piexif
17:13:40-780502 INFO     Installing package: psutil
17:13:42-288157 INFO     Installing package: pyyaml
17:13:43-826701 INFO     Installing package: resize-right
17:13:45-381233 INFO     Installing package: rich
17:13:47-015490 INFO     Installing package: safetensors
17:13:48-586670 INFO     Installing package: scipy
17:13:50-181675 INFO     Installing package: tb_nightly
17:13:52-166273 INFO     Installing package: toml
17:13:53-802559 INFO     Installing package: torchdiffeq
17:13:55-795422 INFO     Installing package: voluptuous
17:13:57-306516 INFO     Installing package: yapf
17:13:58-850310 INFO     Installing package: scikit-image
17:14:00-486388 INFO     Installing package: basicsr
17:14:02-139528 INFO     Installing package: compel==2.0.2
17:14:21-919539 INFO     Installing package: fasteners
17:14:23-853320 INFO     Installing package: typing-extensions==4.7.1
17:14:26-715424 INFO     Installing package: antlr4-python3-runtime==4.9.3
17:14:28-624550 INFO     Installing package: requests==2.31.0
17:14:30-539788 INFO     Installing package: tqdm==4.66.1
17:14:32-414637 INFO     Installing package: accelerate==0.20.3
17:14:34-913982 INFO     Installing package: opencv-python-headless==4.7.0.72
17:14:37-575732 INFO     Installing package: diffusers==0.20.2
17:14:42-398919 INFO     Installing package: einops==0.4.1
17:14:44-337982 INFO     Installing package: gradio==3.43.2
17:15:09-961587 INFO     Installing package: huggingface_hub==0.17.1
17:15:13-419954 INFO     Installing package: numexpr==2.8.4
17:15:15-620220 INFO     Installing package: numpy==1.24.4
17:15:24-166294 INFO     Installing package: numba==0.57.1
17:15:35-541826 INFO     Installing package: pandas==1.5.3
17:15:52-633499 INFO     Installing package: protobuf==3.20.3
17:15:54-786420 INFO     Installing package: pytorch_lightning==1.9.4
17:16:02-385953 INFO     Installing package: transformers==4.31.0
17:16:22-009534 INFO     Installing package: tomesd==0.1.3
17:16:24-299855 INFO     Installing package: urllib3==1.26.15
17:16:26-861126 INFO     Installing package: Pillow==9.5.0
17:16:30-440894 INFO     Installing package: timm==0.9.7
17:16:32-754305 INFO     Installing package: pydantic==1.10.11
17:16:35-653205 INFO     Verifying packages
17:16:35-654205 INFO     Installing package: git+https://github.com/openai/CLIP.git
17:16:42-122700 INFO     Installing package:
                         git+https://github.com/patrickvonplaten/invisible-watermark.git@remove_onnxruntime_depedency
17:16:49-865751 INFO     Installing package: onnxruntime==1.15.1
17:16:54-589640 INFO     Installing package: pi-heif
17:16:57-336011 INFO     Installing package: tensorflow==2.13.0
17:17:44-959573 INFO     Installing package: git+https://github.com/google-research/torchsde
17:17:56-745018 INFO     Extensions: disabled=[]
17:17:56-747018 INFO     Extensions: enabled=['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser',
                         'stable-diffusion-webui-rembg', 'SwinIR'] extensions-builtin
17:17:56-756538 INFO     Extensions: enabled=[] extensions
17:17:56-768059 INFO     Extension preload: {'extensions-builtin': 0.01, 'extensions': 0.0}
17:17:56-872666 WARNING  Incompatible torch version 2.0.1 for ipex windows, reinstalling to 2.0.0a0
17:17:56-874669 WARNING  Package wrong version: torch 2.0.1 required 2.0.0a0
17:17:56-876668 INFO     Installing package: torch==2.0.0a0 intel_extension_for_pytorch==2.0.110+gitba7f6c1 -f
                         https://developer.intel.com/ipex-whl-stable-xpu
17:18:29-812943 WARNING  [WinError 193] %1 is not a valid Win32 application. Error loading
                         "D:\automatic\venv\lib\site-packages\torch\lib\backend_with_compiler.dll" or one of its
                         dependencies.
17:18:29-815944 INFO     Command line args: ['--use-ipex'] use_ipex=True
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ D:\automatic\launch.py:213 in <module>                                                                               │
│                                                                                                                      │
│   212                                                                                                                │
│ ❱ 213     uv, instance = start_server(immediate=True, server=None)                                                   │
│   214     while True:                                                                                                │
│                                                                                                                      │
│ D:\automatic\launch.py:147 in start_server                                                                           │
│                                                                                                                      │
│   146     get_custom_args()                                                                                          │
│ ❱ 147     module_spec.loader.exec_module(server)                                                                     │
│   148     uvicorn = None                                                                                             │
│ in exec_module:883                                                                                                   │
│ in _call_with_frames_removed:241                                                                                     │
│                                                                                                                      │
│ D:\automatic\webui.py:11 in <module>                                                                                 │
│                                                                                                                      │
│    10 from threading import Thread                                                                                   │
│ ❱  11 import modules.loader                                                                                          │
│    12 import torch # pylint: disable=wrong-import-order                                                              │
│                                                                                                                      │
│ D:\automatic\modules\loader.py:10 in <module>                                                                        │
│                                                                                                                      │
│    9 logging.getLogger("DeepSpeed").disabled = True                                                                  │
│ ❱ 10 import torch # pylint: disable=C0411                                                                            │
│   11 try:                                                                                                            │
│                                                                                                                      │
│ D:\automatic\venv\lib\site-packages\torch\__init__.py:133 in <module>                                                │
│                                                                                                                      │
│    132                 err.strerror += f' Error loading "{dll}" or one of its dependencies.'                         │
│ ❱  133                 raise err                                                                                     │
│    134                                                                                                               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
OSError: [WinError 193] %1 is not a valid Win32 application. Error loading
"D:\automatic\venv\lib\site-packages\torch\lib\backend_with_compiler.dll" or one of its dependencies.
```

### Backend

Original

### Model

SD 1.5

### Acknowledgements

- [X] I have read the above and searched for existing issues
- [X] I confirm that this is classified correctly and its not an extension issue

**Comment by Disty0** [comment]
Install Intel OneAPI drivers.

**Comment by jacob1218** [comment]
That stopped the initial error, but now i'm just getting this.
```Using VENV: D:\automatic\venv
14:46:05-324841 INFO     Starting SD.Next
14:46:05-335838 INFO     Python 3.10.11 on Windows
14:46:05-642881 INFO     Version: app=sd.next updated=2023-09-20 hash=89ba8e3c
                         url=https://github.com/vladmandic/automatic.git/tree/master
14:46:06-185069 INFO     Latest published version: 49d181b8a3a24a73400c4f78afe8f1b922bf7b80 2023-09-27T21:26:53Z
14:46:06-195071 INFO     Platform: arch=AMD64 cpu=AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD system=Windows
                         release=Windows-10-10.0.22621-SP0 python=3.10.11
14:46:06-198071 INFO     Intel OneAPI Toolkit detected
14:46:06-221070 ERROR    Intel OneAPI Toolkit is not activated! Activate OneAPI manually!
14:46:06-670510 INFO     Extensions: disabled=[]
14:46:06-672505 INFO     Extensions: enabled=['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser',
                         'stable-diffusion-webui-rembg', 'SwinIR'] extensions-builtin
14:46:06-686449 INFO     Extensions: enabled=[] extensions
14:46:06-688452 INFO     No changes detected: Quick launch active
14:46:06-689453 INFO     Verifying requirements
14:46:06-705451 WARNING  Package wrong version: typing-extensions 4.5.0 required 4.7.1
14:46:06-706814 INFO     Installing package: typing-extensions==4.7.1
14:46:13-557616 WARNING  Package wrong version: numpy 1.24.3 required 1.24.4
14:46:13-558617 INFO     Installing package: numpy==1.24.4
14:46:22-917441 WARNING  Package wrong version: urllib3 2.0.5 required 1.26.15
14:46:22-919440 INFO     Installing package: urllib3==1.26.15
14:46:26-132317 INFO     Verifying packages
14:46:26-135320 INFO     Extensions: disabled=[]
14:46:26-137317 INFO     Extensions: enabled=['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'stable-diffusion-webui-images-browser',
                         'stable-diffusion-webui-rembg', 'SwinIR'] extensions-builtin
14:46:26-147319 INFO     Extensions: enabled=[] extensions
14:46:26-163995 INFO     Extension preload: {'extensions-builtin': 0.01, 'extensions': 0.0}
14:46:26-166999 INFO     Command line args: ['--use-ipex'] use_ipex=True
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ D:\automatic\launch.py:213 in <module>                                                                               │
│                                                                                                                      │
│   212                                                                                                                │
│ ❱ 213     uv, instance = start_server(immediate=True, server=None)                                                   │
│   214     while True:                                                                                                │
│                                                                                                                      │
│ D:\automatic\launch.py:147 in start_server                                                                           │
│                                                                                                                      │
│   146     get_custom_args()                                                                                          │
│ ❱ 147     module_spec.loader.exec_module(server)                                                                     │
│   148     uvicorn = None                                                                                             │
│ in exec_module:883                                                                                                   │
│ in _call_with_frames_removed:241                                                                                     │
│                                                                                                                      │
│ D:\automatic\webui.py:11 in <module>                                                                                 │
│                                                                                                                      │
│    10 from threading import Thread                                                                                   │
│ ❱  11 import modules.loader                                                                                          │
│    12 import torch # pylint: disable=wrong-import-order                                                              │
│                                                                                                                      │
│ D:\automatic\modules\loader.py:10 in <module>                                                                        │
│                                                                                                                      │
│    9 logging.getLogger("DeepSpeed").disabled = True                                                                  │
│ ❱ 10 import torch # pylint: disable=C0411                                                                            │
│   11 try:                                                                                                            │
│                                                                                                                      │
│ D:\automatic\venv\lib\site-packages\torch\__init__.py:133 in <module>                                                │
│                                                                                                                      │
│    132                 err.strerror += f' Error loading "{dll}" or one of its dependencies.'                         │
│ ❱  133                 raise err                                                                                     │
│    134                                                                                                               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
OSError: [WinError 126] The specified module could not be found. Error loading
"D:\automatic\venv\lib\site-packages\torch\lib\backend_with_compiler.dll" or one of its dependencies.```

**Comment by Disty0** [comment]
It's missing libuv.

Follow this guide:

https://github.com/vladmandic/automatic/discussions/2023#discussioncomment-7092173
